src/main/scala/com/typesafe/sbt/web/SbtWeb.scala

```diff
@@ -318,16 +298,34 @@ object SbtWeb extends AutoPlugin {
     nodeModules := nodeModuleGenerators(_.join).map(_.flatten).value
   )
 
+  def webJarsPathPrefix: Def.Initialize[Task[String]] = Def.task {
+    path(s"$WEBJARS_PATH_PREFIX/${moduleName.value}/${version.value}/")
+  }
+
+  def syncExportedAssets(): Def.Initialize[Task[File]] = Def.task {
+    val syncTargetDir = classDirectory.value
+    val syncRequired = TrackLevel.intersection(trackInternalDependencies.value, exportToInternal.value) match {
+      case TrackLevel.TrackAlways =>
+        true
+      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
+        !(syncTargetDir / webJarsPathPrefix.value).exists()
+    }
+    if (syncRequired) {
+      state.value.log.info(s"Exporting Assets for ${moduleName.value}")
+      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
```

Comment:
In other words, this should become something like

```scala
Def.taskDyn {
  ....
  if (syncRequired) Def.task { 
    state.value.log.info(s"Exporting Assets for ${moduleName.value}")
    syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
  }
  else Def.task { syncTargetDir }
}
```

Hopefully that would prevent `exportedMappings.value` getting called when `syncRequired` is false.